src/ui/wxWidgets/DragBarCtrl.cpp

```diff
@@ -96,78 +95,211 @@ struct _DragbarElementInfo {
 // Drag and drop tree or item is last element in drag bar
 #define DND_IDX (NumberOf(DragbarElements) - 1)    // Last entry is DnD
 
-DragBarCtrl::DragBarCtrl(PasswordSafeFrame* frame) : DragBarGenericCtrl(frame, this), m_frame(frame)
+DragBarCtrl::DragBarCtrl(wxWindow *parent, wxWindowID id, const wxPoint &position, const wxSize &size, long style) : wxAuiToolBar(parent, id, position, size, style)
+{
+  CreateToolbar();
+  Bind(wxEVT_AUITOOLBAR_BEGIN_DRAG, &DragBarCtrl::OnDrag, this);
+  Bind(wxEVT_UPDATE_UI, &DragBarCtrl::OnUpdateUI, this, DRAGBAR_TOOLID_BASE, DRAGBAR_TOOLID_BASE + DND_IDX);
+}
+
+DragBarCtrl::~DragBarCtrl()
 {
-  RefreshButtons();
+  Unbind(wxEVT_UPDATE_UI, &DragBarCtrl::OnUpdateUI, this, DRAGBAR_TOOLID_BASE, DRAGBAR_TOOLID_BASE + DND_IDX);
+  Unbind(wxEVT_AUITOOLBAR_BEGIN_DRAG, &DragBarCtrl::OnDrag, this);
 }
 
-void DragBarCtrl::RefreshButtons()
+void DragBarCtrl::CreateToolbar()
 {
   const bool newButtons = PWSprefs::GetInstance()->GetPref(PWSprefs::UseNewToolbar);
 
-#define BTN newButtons? wxBitmap(DragbarElements[idx].bitmap) : wxBitmap(DragbarElements[idx].classic_bitmap)
-#define BTN_DISABLED newButtons? wxBitmap(DragbarElements[idx].bitmap_disabled): wxBitmap(DragbarElements[idx].classic_bitmap_disabled)
+#define BUTTON_ENABLED newButtons ? \
```

Comment:
Preprocessor macros ?!? At least idx should be passed as a parameter to increase readability.
Inline functions would be much better, IMHO.

Reply:
Resolved in fe24e45.